packages/gatsby/cache-dir/loader.js

```diff
@@ -141,6 +142,21 @@ export class BaseLoader {
             throw new Error(`not a valid pageData response`)
           }
 
+          // Handle the page not being in the bundle yet.
+          if (
+            process.env.NODE_ENV === `development` &&
+            jsonPayload.notInDevBundle
+          ) {
+            // Tell the server the user wants to visit this page
+            // to trigger it including the page component's code in the
+            // commons bundles.
+            tellServerWantToVisitPage(jsonPayload.componentChunkName)
+
+            return new Promise(resolve =>
+              setTimeout(() => resolve(this.fetchPageDataJson(loadObj)), 100)
+            )
+          }
```

Comment:
Can we instead of adding `notInDevBundle` field, have the syncRequires have all the keys that it would have normally, just instead of importing actual page component, have something like `Symbol("not-yet-compiled")` and check for that and that would be trigger to tell dev server to compile it? That would not "taint" the `page-data` files with dev-only metadata

Not sure if `Symbol` would work here nicely, but even some hardcoded string could be used instead

Reply:
ahhh yeah I think this would work! I'll try this